src/Features/Core/Portable/UnifiedSuggestions/UnifiedSuggestedActionsSource.cs

```diff
@@ -210,6 +215,14 @@ static CodeFixGroupKey GetGroupKey(CodeFix fix)
             using var fixAllSuggestedActionsDisposer = ArrayBuilder<IUnifiedSuggestedAction>.GetInstance(out var fixAllSuggestedActions);
             foreach (var scope in supportedScopes)
             {
+                if (scope is FixAllScope.ContainingMember or FixAllScope.ContainingType)
```

Comment:
This is the only real change in this file, rest all is making the entire call chain async. We bail out from showing ContainingMember or ContainingType fix all option if we cannot find a containing member or type respectively.

Reply:
open question.  how does TS/F# play into this?  is finding the containing type/member part of some API they can imeplement?

Reply:
That is outside the scope of this PR. If we want to add support for these scopes in TS/F#, we would need some way to be able to query them to get the containing member span for trigger position.

Reply:
I actually went ahead and did this refactoring in this PR itself with https://github.com/dotnet/roslyn/pull/59989/commits/64baa64648a00220629e760d343a8b237728a9be. Other languages can now implement the `IFixAllSpanMappingService` language service to enable fix all for these new scopes.